core/src/main/scala/org/apache/spark/sql/delta/implicits/package.scala

```diff
@@ -16,104 +16,33 @@
 
 package org.apache.spark.sql.delta
 
-import org.apache.spark.sql.catalyst.expressions.Expression
-import org.apache.spark.sql.catalyst.plans.QueryPlan
-import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
-import org.apache.spark.sql.catalyst.rules.{RuleId, UnknownRuleId}
-import org.apache.spark.sql.catalyst.trees.{AlwaysProcess, TreePatternBits}
-import org.apache.spark.sql.types.{ArrayType, MapType, StructField, StructType}
+import org.apache.spark.sql.delta.actions.AddFile
+import org.apache.spark.sql.delta.implicits.RichSparkClasses
+import org.apache.spark.sql.delta.util.DeltaEncoders
 
-package object implicits {
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
 
-  /**
-   * This implicit class is used to provide helpful methods used throughout the code that are not
-   * provided by Spark-Catalyst's StructType.
-   */
-  implicit class RichStructType(structType: StructType) {
+package object implicits extends DeltaEncoders with RichSparkClasses {
+  // Define a few implicit classes to provide the `toDF` method. These classes are not using generic
+  // types to avoid touching Scala reflection.
+  implicit class RichAddFileSeq(files: Seq[AddFile]) {
+    def toDF(spark: SparkSession): DataFrame = spark.implicits.localSeqToDatasetHolder(files).toDF
 
-    /**
-     * Returns a field in this struct and its child structs, case insensitively.
-     *
-     * If includeCollections is true, this will return fields that are nested in maps and arrays.
-     *
-     * @param fieldNames The path to the field, in order from the root. For example, the column
-     *                   nested.a.b.c would be Seq("nested", "a", "b", "c").
-     */
-    def findNestedFieldIgnoreCase(
```

Comment:
is it safe to delete these methods?

Reply:
nvm. I see it's added in RichSparkClasses. sgtm.

Reply:
These classes are moved to `RichSparkClasses.scala`